src/util.py

```diff
@@ -260,12 +265,156 @@ def xr_add_cyclic_longitudes(da, coord):
 
     return new_da
 
+
+def findLocalExtrema(da, highVal=1040, lowVal=975, eType='Low'):
+    """
+    Utility function to find local low/high field variable coordinates on a contour map. To classify as a local high, the data
+    point must be greater than highVal, and to classify as a local low, the data point must be less than lowVal.
+
+    Args:
+
+        da: (:class:`xarray.DataArray`):
+            Xarray data array containing the lat, lon, and field variable (ex. pressure) data values
+
+        highVal (:class:`int`):
+            Data value that the local high must be greater than to qualify as a "local high" location.
+            Default highVal is a pressure value of 1040 hectopascals.
+
+        lowVal (:class:`int`):
+            Data value that the local low must be less than to qualify as a "local low" location.
+            Default lowVal is a pressure value of 975 hectopascals.
+
+        eType (:class:`str`):
+            'Low' or 'High'
+            Determines which extrema are being found- minimum or maximum, respectively.
+            Default eType is 'Low'.
+
+    Returns:
+
+        clusterExtremas (:class:`list`):
+            List of coordinate tuples in GPS form (lon in degrees, lat in degrees)
+            that specify local low/high locations
+
+    """
+    import numpy as np
+    from sklearn.cluster import DBSCAN
+    import warnings
+
+    # Create a 2D array of coordinates in the same shape as the field variable data
+    # so each coordinate is easily mappable to a data value
+    coordarr = []
+    for y in np.array(da.lat):
```

Comment:
Please search for `numpy.meshgrid, and you could solve this task with it because using for-loops are not the most efficient way usually in Python or numpy, especially.